src/transformers/modeling_utils.py

```diff
@@ -1457,10 +1475,14 @@ def _load_state_dict_into_model(
 
         if _fast_init:
             # retrieve unintialized modules and initialize
-            unintialized_modules = model.retrieve_modules_from_names(
+            uninitialized_modules = model.retrieve_modules_from_names(
                 missing_keys, add_prefix=add_prefix, remove_prefix=remove_prefix
             )
-            for module in unintialized_modules:
+            for module in uninitialized_modules:
+                # # XXX: untested
+                # # put the params that aren't going to be replaced by state_dict to CPU before running init
+                # for p in module.parameters(recursive=False):
+                #     p = p.to(device='cpu')
```

Comment:
Does this need clean up or is it a future direction to explore?

Reply:
a missed leftover - thank you! fixed!